realsense2_camera/launch/rs_rgbd.launch

```diff
@@ -179,14 +186,34 @@ Processing enabled by this node:
     </group>
 
     <group if="$(eval depth_registered_processing and hw_registered_processing)">
-      <!-- Publish registered XYZRGB point cloud with hardware registered input (ROS Realsense depth alignment) -->
-      <node pkg="nodelet" type="nodelet" name="points_xyzrgb_hw_registered"
-            args="load depth_image_proc/point_cloud_xyzrgb $(arg manager) $(arg bond)" respawn="$(arg respawn)">
-        <remap from="rgb/image_rect_color"        to="$(arg rgb)/image_rect_color" />
-        <remap from="rgb/camera_info"             to="$(arg rgb)/camera_info" />
-        <remap from="depth_registered/image_rect" to="$(arg depth_registered)/image_raw" />
-        <remap from="depth_registered/points"     to="$(arg depth_registered_pub)/points" />
-      </node>
+      <group if="$(arg pc_rgb)">
+        <!-- Publish registered XYZRGB point cloud with hardware registered input (ROS Realsense depth alignment) -->
+        <node pkg="nodelet" type="nodelet" name="points_xyzrgb_hw_registered"
+              args="load depth_image_proc/point_cloud_xyzrgb $(arg manager) $(arg bond)" respawn="$(arg respawn)">
+          <remap from="rgb/image_rect_color"        to="$(arg rgb)/image_rect_color" />
+          <remap from="rgb/camera_info"             to="$(arg rgb)/camera_info" />
+          <remap from="depth_registered/image_rect" to="$(arg depth_registered)/image_raw" />
+          <remap from="depth_registered/points"     to="$(arg depth_registered_pub)/points" />
+        </node>
+      </group>
+
+      <group if="$(arg pc_ir)">
+        <!-- Convert IR from mono to color image -->
+        <node pkg="realsense2_camera" type="mono_to_color.py" name="mono_to_color" respawn="$(arg respawn)">
```

Comment:
How much processing power does this take, and do we need it?

Reply:
I tried making the pointcloud with the 1-channel mono IR image but the pointcloud generation nodelet threw some errors. If we can figure out how to make an XYZI pointcloud instead of XYZRGB it could work. But the alvar marker detector node must also be able to accept the XYZI pointcloud